### PR TITLE
feat: IPython integration with ABI-related constructs

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

fixes: #960 

requires https://github.com/ApeWorX/ethpm-types/pull/133/files

Adds IPython pretty repr and `.info` properties to the following:

* Method handlers, both call and tx
* Contract events
* Custom error types (fingers crossed on this one, may not work because they are types and not instances)

### How I did it

tbd

### How to verify it

`ape console`

```
contract.myMethod
contract.MyEvent
contract.MyError
```

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: src/ape/contracts/base.py [hunk 1]
```diff
@@ -149,11 +151,52 @@ def __repr__(self) -> str:
...
         return f"{self.contract.__repr__()}.{self.abis[-1].name}"
 
+    @log_instead_of_fail()
+    def _repr_pretty_(self, printer, cycle):
+        """
+        Show the NatSpec of a Method in any IPython console (including ``ape console``).
+        """
+        if info := self.info:
+            text = Text(info)
+            rich_print(f"<natspec>\n{info}")
```
Comment (fubuloubu) [inline src/ape/contracts/base.py:161]: Whats the `<natspec>` tag do? Or is this just debugging for now?

Comment (antazoey) [reply src/ape/contracts/base.py:161]: I am trying to think of a way to get all the info starting on a newline after `Out[1]:` part or else the indentation seems weird on the first line

**Comment by antazoey** [comment]
Some examples!

A method:

Screenshot 2024-07-11 at 10 51 20

An event:

Screenshot 2024-07-11 at 10 51 37

A custom error:

Screenshot 2024-07-11 at 10 52 01

**Comment by antazoey** [comment]
as always, I am only getting a failure in CI and not locally, so debugging this PITA

**Review by fubuloubu — DISMISSED** [review]

**Pull request merged**